app/models/services/service_credential_binding_view.rb

```diff
@@ -85,11 +85,111 @@ module Types
 
       class View < Sequel::Model(VIEW)
         plugin :single_table_inheritance,
-               :type,
-               model_map: {
-                 'app' => 'VCAP::CloudController::ServiceBinding',
-                 'key' => 'VCAP::CloudController::ServiceKey'
-               }
+          :type,
+          model_map: {
+            'app' => 'VCAP::CloudController::ServiceBinding',
+            'key' => 'VCAP::CloudController::ServiceKey'
+          }
+
+        # Custom eager loading: https://github.com/jeremyevans/sequel/blob/master/doc/advanced_associations.rdoc#label-Custom+Eager+Loaders
+        many_to_one :service_instance_sti_eager_load,
+          dataset: -> { raise 'Must be used for eager loading' },
+          eager_loader_key: nil, # set up id_map ourselves
+          eager_loader: proc { |eo|
+            service_instance_id_to_keys = {}
+            service_instance_guid_to_bindings = {}
+            eo[:rows].each do |scb|
+              scb.associations[:service_instance] = nil
+              case scb # keys are joined by ID and bindings by GUID
+              when ServiceKey
+                service_instance_id_to_keys[scb.service_instance_id] ||= []
+                service_instance_id_to_keys[scb.service_instance_id] << scb
+              when ServiceBinding
+                service_instance_guid_to_bindings[scb.service_instance_guid] ||= []
+                service_instance_guid_to_bindings[scb.service_instance_guid] << scb
+              end
+            end
+            ds = ServiceInstance.where(id: service_instance_id_to_keys.keys).or(guid: service_instance_guid_to_bindings.keys)
```

Comment:
I am not sure how efficient is this. It seems that Postgresql can do a "bitmap or" of the two indexes, and wouldn't be a big issue. Not sure about other databases. Just something to keep an eye on.